adam_core/utils/tests/test_indexable.py

```diff
@@ -0,0 +1,470 @@
+    array = np.concatenate([np.arange(0, 5), np.arange(0, 5)])
+
+    with pytest.raises(ValueError):
+        _convert_grouped_array_to_slices(array)
+
+
+def test__convert_grouped_array_to_slices():
+
+    # Test that the function correctly converts a sorted array to slices
+    array = np.arange(0, 10)
+    slices = _convert_grouped_array_to_slices(array)
+    desired = np.array(
+        [
+            slice(0, 1, 1),
+            slice(1, 2, 1),
+            slice(2, 3, 1),
+            slice(3, 4, 1),
+            slice(4, 5, 1),
+            slice(5, 6, 1),
+            slice(6, 7, 1),
+            slice(7, 8, 1),
+            slice(8, 9, 1),
+            slice(9, 10, 1),
+        ]
+    )
+    np.testing.assert_equal(slices, desired)
+
+    # The order of the sorted array should not matter as long as all elements that
+    # are supposed to be in the same slice are grouped together
+    array = np.arange(0, 10)[::-1]
+    slices = _convert_grouped_array_to_slices(array)
+    np.testing.assert_equal(slices, desired)
+
+    # Test that the function correctly converts a sorted array to slices
+    # this time with duplicates in the array
+    array = np.concatenate([np.arange(0, 5), np.arange(0, 5)])
+    array.sort()
+    slices = _convert_grouped_array_to_slices(array)
+    desired = np.array(
+        [
+            slice(0, 2, 1),
+            slice(2, 4, 1),
+            slice(4, 6, 1),
+            slice(6, 8, 1),
+            slice(8, 10, 1),
+        ]
+    )
+    np.testing.assert_equal(slices, desired)
+
+    # Test that the function correctly converts a sorted array to slices
+    # this time with only some duplicates in the array
+    array = np.concatenate([np.arange(0, 2), np.arange(0, 2), np.arange(0, 5)])
+    array.sort()
+    slices = _convert_grouped_array_to_slices(array)
+    desired = np.array(
+        [
+            slice(0, 3, 1),
+            slice(3, 6, 1),
+            slice(6, 7, 1),
+            slice(7, 8, 1),
+            slice(8, 9, 1),
+        ]
+    )
+    np.testing.assert_equal(slices, desired)
+
+
+def test__map_values_to_integers():
+
+    # Test that the function correctly maps integer values to integers
+    values = np.arange(0, 10)
+    mapping = _map_values_to_integers(values)
+    desired = np.arange(0, 10)
+    np.testing.assert_equal(mapping, desired)
+
+    # Test that the function correctly maps string values to integers
+    values = np.array(["a", "a", "b", "c", "d", "e", "e", "e"])
+    mapping = _map_values_to_integers(values)
+    desired = np.array([0, 0, 1, 2, 3, 4, 4, 4])
+    np.testing.assert_equal(mapping, desired)
+
+    # Test that the function correctly maps float values with duplicates to integers
+    values = np.concatenate([np.arange(0, 10, dtype=float) for i in range(2)])
+    mapping = _map_values_to_integers(values)
+    desired = np.concatenate([np.arange(0, 10), np.arange(0, 10)])
+    np.testing.assert_equal(mapping, desired)
```

Comment:
Does `_map_values_to_integers` work if the values are in a scrambled order?

```
values = np.array(["a", "b", "a"])
mapping = _map_values_to_integers(values)
desired = np.array([0, 1, 0]) # maybe???
np.testing.assert_equal(mapping, desired)
```

Reply:
Yep -- that's actually what this test is testing! I'll add your exact example as an additional test for extra clarity.